psealock/gutenberg
# Make nav components more atomic

## Description
Breaks down nav components into smaller pieces so that the nav itself is more customizable.

This is in need of testing and there are still some outstanding issues with accessibility, but prefer to handle these in follow-ups to keep the scope of this PR down.

This PR tries to strike a balance between flexibility and ease of use.  I think in keeping this component as presentational as possible, it would be ideal to have the consumer map over and use navigational components directly:

```js
<NavigationMenu>
  <NavigationMenuItem
    onClick={ ... }
  >
     Menu Item
  </NavigationMenuItem>
<NavigationMenu />
```

instead of passing an array of items:

```js
<Navigation active={ active } items={ items }>
  <NavigationMenu />
</Navigation>
```

However, using the former option means putting more responsibility on the consumer to filter the array of items.  Open to feedback here and whether or not there's a way to improve this.

## Screenshots 
<img width="289" alt="Screen Shot 2020-07-28 at 6 08 57 PM" src="https://user-images.githubusercontent.com/10561050/88685367-8c292a00-d0fe-11ea-8ed0-eef9a353090d.png">


## Testing

1. Run `npm run storybook:dev`
1. Check the `Navigation` component.
1. Check that items are rendered, navigation, and back buttons work, and the navigation title shows the correct title.
